tests/foreman/api/test_puppetmodule.py

```diff
@@ -19,28 +19,30 @@
 
 from robottelo.constants import PUPPET_MODULE_NTP_PUPPETLABS
 from robottelo.helpers import get_data_file
-from robottelo.test import APITestCase
 
 
-class RepositorySearchTestCase(APITestCase):
-    """Tests that search for puppet modules and filter by repository."""
+@pytest.fixture(scope='module')
+def make_product_repo():
+    product = entities.Product().create()
+    repo = entities.Repository(content_type='puppet', product=product).create()
+    return {'product': product, 'repo': repo}
```

Comment:
Its a good pattern to cleanup the resources created by the fixture, which can be done by using `yield` and then deleting the created entities.

Reply:
I just didn't want to step further than the original test, but used the yield/delete in the end.